File: Week_5/1991. 트리 순회/트리 순회.cs
using System;

class Program
{
    static char[,] parent = new char[26,2];
    
    static void preorder(char root)
    {
        if(root == '.') return;
        else
        {
            Console.Write(root);
            preorder(parent[root-'A',0]);
            preorder(parent[root-'A',1]);
        }
    }
    
    static void inorder(char root)
    {
        if(root == '.') return;
        else
        {
            inorder(parent[root - 'A',0]);
            Console.Write(root);
            inorder(parent[root - 'A',1]);
        }
    }
    static void postorder(char root)
    {
        if(root == '.') return;
        else 
        {
            postorder(parent[root - 'A',0]);
            postorder(parent[root - 'A',1]);
            Console.Write(root);
        }
    }
    static void Main() {
        int n = int.Parse(Console.ReadLine());
        
        for(int i=0;i<n;i++)
        {
            string[] s = Console.ReadLine().Split(' ');
            
            parent[char.Parse(s[0])-'A',0] = char.Parse(s[1]);
            parent[char.Parse(s[0])-'A',1] = char.Parse(s[2]);
        }
        preorder('A');
        Console.WriteLine();
        inorder('A');
        Console.WriteLine();
        postorder('A');
        Console.WriteLine();
    }
}

File: Week_5/1991. 트리 순회/트리 순회.py
parent = [[0 for _ in range(2)] for _ in range(26)]

def preorder(root):
    if root == '.': return
    else:
        print(root, end = "")
        preorder(chr(parent[ord(root)-65][0]))
        preorder(chr(parent[ord(root)-65][1]))

def inorder(root):
    if root == '.': return
    else:
        inorder(chr(parent[ord(root)-65][0]))
        print(root,end = "")
        inorder(chr(parent[ord(root)-65][1]))

def postorder(root):
    if root =='.': return
    else:
        postorder(chr(parent[ord(root)-65][0]))
        postorder(chr(parent[ord(root)-65][1]))
        print(root,end = "")

n = int(input())

for i in range(n):
    x,y,z = input().split()
    parent[ord(x)-65][0] = ord(y)
    parent[ord(x)-65][1] = ord(z)
    
preorder('A')
print()
inorder('A')
print()
postorder('A')